### ISSUE TITLE
in the cloud-update-ci job JJB fails to update ci.opensuse.org

### ISSUE THREAD

**Issue by JanZerebecki** [SUSE-Cloud/automation#2746]
From https://ci.suse.de/job/cloud-update-ci/741/console
```
INFO:jenkins_jobs.builder:Reconfiguring jenkins job openstack-trackupstream
INFO:requests.packages.urllib3.connectionpool:Resetting dropped connection: ci.opensuse.org

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/parallel.py", line 62, in run
    **task['kwargs'])
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 355, in parallel_update_job
    self.update_job(job.name, job.output().decode('utf-8'))
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 139, in update_job
    self.jenkins.reconfig_job(job_name, xml)
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 1223, in reconfig_job
    headers=DEFAULT_HEADERS
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 541, in jenkins_open
    return self.jenkins_request(req, add_crumb, resolve_auth).text
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 560, in jenkins_request
    self._request(req))
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 520, in _response_handler
    "empty response" % self.server)
EmptyResponseException: Error communicating with server[https://ci.opensuse.org/]: empty response
Traceback (most recent call last):
  File "/usr/bin/jenkins-jobs", line 10, in <module>
    sys.exit(main())
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/entry.py", line 158, in main
    jjb.execute()
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/entry.py", line 139, in execute
    ext.obj.execute(self.options, self.jjb_config)
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/subcommand/update.py", line 150, in execute
    existing_only=options.existing_only)
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 340, in update_jobs
    raise result
jenkins.EmptyResponseException: Error communicating with server[https://ci.opensuse.org/]: empty response
```

**Comment by JanZerebecki** [comment]
The error output changed, it is now:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/parallel.py", line 62, in run
    **task['kwargs'])
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 355, in parallel_update_job
    self.update_job(job.name, job.output().decode('utf-8'))
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 136, in update_job
    if self.is_job(job_name):
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 147, in is_job
    if job_name in self.job_list:
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 113, in job_list
    self._job_list = set(job['fullname'] for job in self.jobs)
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 105, in jobs
    self._jobs = self.jenkins.get_all_jobs()
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 1010, in get_all_jobs
    jobs = [(0, "", self.get_info(query=JOBS_QUERY)['jobs'])]
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 753, in get_info
    requests.Request('GET', self._build_url(url))
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 541, in jenkins_open
    return self.jenkins_request(req, add_crumb, resolve_auth).text
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 557, in jenkins_request
    self.maybe_add_crumb(req)
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 352, in maybe_add_crumb
    'GET', self._build_url(CRUMB_URL)), add_crumb=False)
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 541, in jenkins_open
    return self.jenkins_request(req, add_crumb, resolve_auth).text
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 571, in jenkins_request
    raise JenkinsException(msg)
JenkinsException: Error in request. Possibly authentication failed [401]: Invalid password/token for user: jdsnapi
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 401 Invalid password/token for user: jdsnapi</title>
</head>
<body><h2>HTTP ERROR 401</h2>
<p>Problem accessing /crumbIssuer/api/json. Reason:
<pre>    Invalid password/token for user: jdsnapi</pre></p><hr><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.4.z-SNAPSHOT</a><hr/>

</body>
</html>
```

**Comment by JanZerebecki** [comment]
@jdsn Can you fix this?

**Comment by jdsn** [comment]
some fixes are deployed and internal jenkins can get updates again
ci.o.o suffers from a bug in python-python-jenkins - will try to find a workaround and/or fix for upstream

**Comment by JanZerebecki** [comment]
Now the error is back to (nearly the same as in the first comment https://github.com/SUSE-Cloud/automation/issues/2746#issue-360261548 ):
```
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): ci.opensuse.org
INFO:jenkins_jobs.builder:Reconfiguring jenkins job openstack-cleanvm
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/parallel.py", line 62, in run
    **task['kwargs'])
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 355, in parallel_update_job
    self.update_job(job.name, job.output().decode('utf-8'))
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 139, in update_job
    self.jenkins.reconfig_job(job_name, xml)
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 1225, in reconfig_job
    headers=DEFAULT_HEADERS
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 564, in jenkins_open
    return self.jenkins_request(req, add_crumb, resolve_auth).text
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 583, in jenkins_request
    self._request(req))
  File "/usr/lib/python2.7/site-packages/jenkins/__init__.py", line 543, in _response_handler
    "empty response" % self.server)
EmptyResponseException: Error communicating with server[https://ci.opensuse.org/]: empty response
INFO:jenkins_jobs.builder:Reconfiguring jenkins job openstack-devstack-ipv4
INFO:requests.packages.urllib3.connectionpool:Resetting dropped connection: ci.opensuse.org
```
...
```
EmptyResponseException: Error communicating with server[https://ci.opensuse.org/]: empty response
Traceback (most recent call last):
  File "/usr/bin/jenkins-jobs", line 10, in <module>
    sys.exit(main())
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/entry.py", line 165, in main
    jjb.execute()
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/entry.py", line 146, in execute
    ext.obj.execute(self.options, self.jjb_config)
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/cli/subcommand/update.py", line 146, in execute
    existing_only=options.existing_only)
  File "/usr/lib/python2.7/site-packages/jenkins_jobs/builder.py", line 340, in update_jobs
    raise result
jenkins.EmptyResponseException: Error communicating with server[https://ci.opensuse.org/]: empty response
Makefile:66: recipe for target 'cioo_deploy' failed
```

**Comment by JanZerebecki** [comment]
Upstream fix: https://review.openstack.org/#/c/628712/

**Comment by JanZerebecki** [comment]
Upstream fix was included in version 1.5.0 of python-jenkins. Error is gone.
